src/index.js

```diff
@@ -44,7 +46,12 @@ export function createSelectorCreator(memoize, ...memoizeOptions) {
     const memoizedResultFunc = memoize(
       (...args) => {
         recomputations++
-        return resultFunc(...args)
+        try {
+          return resultFunc(...args)
+        } catch (e) {
+          errorSubscriptions.emitError(e, resultFunc, args, dependencies)
```

Comment:
That's a good point.

I don't like the idea of hijacking the Error. However, you are right, it could be possible to create a higher order function like this one:

```
import { createSelector } from 'reselect';

const catchableSelector = (...args) => {
  const selector = createSelector(args);
  return (...state) => {
    try {
      return selector(...state);
    } catch (error) {
      const resultFunc = args.slice(-1);
      let dependencies = args.(0, -1);
      dependencies = dependencies.length === 1 && Array.isArray(dependencies[0]) ?
        ...dependencies[0] :
        dependencies;
      // DO WHATEVER
    }
  }
};

export default catchableSelector;
```

And then in your project you should always use the `catchableSelector` instead of using directly the `createSelector`... It's an option. I am not a big fan of this approach, but it's an option. One inconvenient with this approach is that you wouldn't be able to have access to the parameters that were received by the result function of the selector that crashed. Which is one of the most valuable pieces of information that you would get if this PR got accepted.